app/services/ocean_service.rb

```diff
@@ -0,0 +1,62 @@
+class OceanService
+  ISLAND_SCATTER_RADIUS = 200
+  CENTER = [500,500]
+
+  def initialize(number_of_islands)
+    @number_of_islands = number_of_islands
+  end
+
+  def call
+    { islands: islands }
+  end
+
+  def islands
+    if @number_of_islands.zero? || @number_of_islands.nil?
+      []
+    else
+      size = island_size
+      biggest_island_size = size.sort.last
+      generate_positions(biggest_island_size).each_with_index.map { |position,i|IslandService.new(position, size[1-i]).call}
+    end
+  end
+
+  def distance(edge)
+    pt1 = edge.first
+    pt2 = edge.last
+    mx = (pt2.first - pt1.first)
+    my = (pt2.last - pt1.last)
+    Math.sqrt(mx*mx + my*my)
+  end
+
+  def generate_positions(biggest_island_size)
+    if @number_of_islands == 1
+      [make_point]
+    elsif @number_of_islands == 2
+      begin
+        points = [make_point,make_point]
+      end while (2*biggest_island_size) > distance(points)
+      points
```

Comment:
this part should at least be inside a function with a speaking name 
```
begin
         points = [make_point,make_point]
       end while (2*biggest_island_size) > distance(points)
       points
``` 
also if you can try something that is not using while / begin while or for, since those things are less readable